subsys/bluetooth/audio/has.c

```diff
@@ -1122,48 +1125,51 @@ int bt_has_preset_name_change(uint8_t index, const char *name)
 }
 #endif /* CONFIG_BT_HAS_PRESET_SUPPORT */
 
-static int has_init(const struct device *dev)
+int bt_has_register(const struct bt_has_register_param *param)
 {
-	ARG_UNUSED(dev);
+	int err;
+
+	LOG_DBG("param %p", param);
+
+	CHECKIF(!param) {
+		LOG_DBG("NULL params pointer");
+		return -EINVAL;
+	}
 
 	/* Initialize the supported features characteristic value */
-	has.features = CONFIG_BT_HAS_HEARING_AID_TYPE & BT_HAS_FEAT_HEARING_AID_TYPE_MASK;
+	has.features = param->type;
 
 	if (IS_ENABLED(CONFIG_BT_HAS_PRESET_SUPPORT)) {
 		has.features |= BT_HAS_FEAT_DYNAMIC_PRESETS;
 	}
 
-	if (IS_ENABLED(CONFIG_BT_HAS_HEARING_AID_BINAURAL)) {
-		if (IS_ENABLED(CONFIG_BT_HAS_PRESET_SYNC_SUPPORT)) {
-			has.features |= BT_HAS_FEAT_PRESET_SYNC_SUPP;
+	if (param->preset_sync_support) {
+		if (param->type != BT_HAS_HEARING_AID_TYPE_BINAURAL) {
+			LOG_DBG("Preset sync support only available for binaural hearing aid type");
+			return -EINVAL;
 		}
 
-		if (!IS_ENABLED(CONFIG_BT_HAS_IDENTICAL_PRESET_RECORDS)) {
-			has.features |= BT_HAS_FEAT_INDEPENDENT_PRESETS;
+		has.features |= BT_HAS_FEAT_PRESET_SYNC_SUPP;
+	}
+
+	if (param->independent_presets) {
+		if (param->type != BT_HAS_HEARING_AID_TYPE_BINAURAL) {
+			LOG_DBG("Independent presets only available for binaural hearing aid type");
+			return -EINVAL;
 		}
```

Comment:
The documentation for the `struct bt_has_register_param` should contain these restrictions :)

Reply:
> The documentation for the `struct bt_has_register_param` should contain these restrictions :)

You are right - forgot it ends up in the actual documentation ;)

Reply:
I added a remark on this being for binaural only